xwiki-platform-core/xwiki-platform-mentions/xwiki-platform-mentions-default/src/main/java/org/xwiki/mentions/internal/DefaultMentionNotificationService.java

```diff
@@ -50,12 +50,12 @@
     private EntityReferenceSerializer<String> serializer;
 
     @Override
-    public void sendNotif(DocumentReference authorReference, DocumentReference documentReference,
+    public void sendNotif(String authorReference, String documentReference,
         DocumentReference mentionedIdentity, MentionLocation location, String anchorId)
     {
         MentionEventParams params = new MentionEventParams()
-                                        .setUserReference(authorReference.toString())
-                                        .setDocumentReference(documentReference.toString())
+                                        .setUserReference(authorReference)
```

Comment:
So it was wrong to use `toString` before, now it would still make sense to have the method signature using real reference types, and to use the serializer here.